hw-event-proxy/cmd/main.go

```diff
@@ -0,0 +1,224 @@
+// Copyright 2021 The Cloud Native Events Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"sync"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	hwevent "github.com/redhat-cne/sdk-go/pkg/hwevent"
+
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/pb"
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/restclient"
+	"github.com/redhat-cne/hw-event-proxy/hw-event-proxy/util"
+	"google.golang.org/grpc"
+
+	v1hwevent "github.com/redhat-cne/sdk-go/v1/hwevent"
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	hwEventVersion string = "v1"
+	eventType      string = "HW_EVENT"
+)
+
+var (
+	apiPath         = "/api/cloudNotifications/v1/"
+	apiPort         int
+	json            = jsoniter.ConfigCompatibleWithStandardLibrary
+	pub             pubsub.PubSub
+	resourceAddress string
+)
+
+func main() {
+	flag.IntVar(&apiPort, "api-port", 8080, "The address the rest api endpoint binds to.")
+	flag.Parse()
+	util.InitLogger()
+
+	nodeName := os.Getenv("NODE_NAME")
+	if nodeName == "" {
+		log.Error("cannot find NODE_NAME environment variable,setting to default `mock` node")
+		nodeName = "mock"
+	}
+
+	resourceAddress = fmt.Sprintf("/cluster/node/%s/redfish/event", nodeName)
+
+	var err error
+	pub, err = createPublisher()
+	if err != nil {
+		return
+	}
+	log.Infof("Created publisher %v", pub)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	startWebhook(&wg)
+	log.Info("waiting for events")
+	wg.Wait()
+}
+
+func createPublisher() (pub pubsub.PubSub, err error) {
+	baseURL := types.ParseURI(fmt.Sprintf("http://localhost:%d%s", apiPort, apiPath))
```

Comment:
baseUrl can be a global variable since you are using in more than one place (publish event )